jujugui/static/gui/src/app/app.js

```diff
@@ -757,15 +763,21 @@ YUI.add('juju-gui', function(Y) {
             return;
           }
           const selectedModel = this._pickModel(modelList);
+          const jimmURL = this.get('jimmURL');
+          let socketTemplate = this.get('socketTemplate');
+
+          if (jimmURL) {
+            socketTemplate = `${jimmURL}${socketTemplate}`;
```

Comment:
I am really confused about this: connecting to jimm should be the same as connecting to a controller, so the controllerSocketTemplate should be used. This concatenation of a URL and a template feels wrong, like an hack, and we are doing this hack in two different places in the code, which is also a time bomb. I'd remove all of this, all mentions to jimm, and let the GUI handle the jimm connection with just passing the proper values on the 4 config options that I mentioned above.